### DISCUSSION TITLE
max Distance exceeding circuit length

### DISCUSSION THREAD

**Discussion by harish5p** [theOehrly/Fast-F1#192] [Q&A]
```
F1Session = ff1.get_session(year, race, session)
laps = F1Session.load_laps(with_telemetry=True)

#Picks The Laps From The Inputted Driver
driverLaps = laps.pick_driver(driver)

#Creates A Dataframe For The Drivers Lap Info
telemetry = pd.DataFrame()
#Loops Through Every Lap In The Race
for lap in driverLaps.iterlaps():
    #Gets The Laps Distance
    driver_telemetry = lap[1].get_telemetry().add_distance()
    #Gets The Laps Number
    driver_telemetry['Lap'] = lap[1]['LapNumber']

    #Adds It To The Total Telemetry
    telemetry = telemetry.append(driver_telemetry)

#Sets The Amount Of MiniSectors
if msCount is None:
    #The Default Is 25
    numberOfMiniSectors = 25
else:
    numberOfMiniSectors = msCount

#The Laps Total Distance
totalLapDistance = max(telemetry['Distance'])

# Generate equally sized mini-sectors
miniSectorLength = totalLapDistance / numberOfMiniSectors

#Creates A MiniSector Array
miniSectors = [0]
```

For 
```
driver = 'RUS'
year = 2022
race = 9
session = 'FP2'
msCount = 50
```

 On Lap 7 for 'RUS' ```telemetry['Distance']``` started from 0 but ```max(telemetry['Distance'])```  is 4785.16, but if we take 'VER' telemetry it is around  4433. [The official circuit length is 4361m](https://www.formula1.com/en/racing/2022/Canada/Circuit.html)

Is it because of integrating error as [mentioned in the documentation](https://theoehrly.github.io/Fast-F1/core.html#fastf1.core.Telemetry.add_distance)?

**Comment by theOehrly** [comment]
That difference seems way too large to be caused by integration error. Is there anything special about that lap? I'd need to take a closer look at the data first to figure out what causes this difference.

**Reply by harish5p** [reply]
I'm not sure what happened during that lap, only watched highlights for qualification. If possible, also look into Norris and Bottas laps, they might help. When I was calculating fastest potential lap times based on mini sectors. RUS, NOR, BOT fastest potential lap times were slower than their actual lap times. Later, looking for error, I noticed this RUS Lap 7 issue.

**Reply by theOehrly** [reply]
I spent a bit of time trying to figure out what is going on. It looks pretty much like the source data is wrong. I'm not 100% sure what the reason is, but I can make a guess. 
Lap 7 was an inlap for Russel. The end of the inlap is triggered at the end of the pit lane. So after he was in the garage for multiple minutes, when he drives out again. This is normal. It just depends on the position of the timing line.
When reviewing all the data, also for the duration when he was inside the garage, it looks like Mercedes fired up the engine in the garage again. Then they went up through all the gears once. Somehow, the telemetry also show a (very jittery) speed here. It's pretty much impossible, that Russel was actually driving though. So I guess this testing of the car in the garage is mistakenly interpreted as Russel driving. There's no simple fix for this in FastF1.
